File: app/src/main/java/applab/veiligthuis/repository/MeldingRepository.java
package applab.veiligthuis.repository;

import androidx.lifecycle.LiveData;

import java.util.List;

import applab.veiligthuis.model.Melding;

public interface MeldingRepository {

    LiveData<List<Melding>> getMeldingenListLiveData();
    LiveData<Melding> getMeldingById(String id);

}

File: app/src/main/java/applab/veiligthuis/data/melding/MeldingRepository.kt
package applab.veiligthuis.data.melding



import applab.veiligthuis.domain.model.melding.Melding
import applab.veiligthuis.domain.util.MeldingType
import kotlinx.coroutines.flow.Flow

interface MeldingRepository {
    fun getMeldingen(paths: List<String>, meldingType: MeldingType): Flow<List<Melding?>>
    fun getMelding(meldingKey: String, meldingType: MeldingType): Flow<Melding>

    @Throws(MeldingInsertException::class)
    fun insertOrUpdateMelding(melding: Melding)
    fun deleteMelding(melding: Melding)
}